sdk/communication/Azure.Communication.CallAutomation/tests/Events/CallAutomationEventParserTests.cs

```diff
@@ -506,6 +507,7 @@ public void RecognizeCompletedEventParsed_Test()
                 operationContext: "operationContext",
                 recognitionType: CallMediaRecognitionType.Dtmf,
                 collectTonesResult: new CollectTonesResult(new DtmfTone[] { DtmfTone.Five }),
```

Comment:
This test is logic wrong, if the  `recognitionType: CallMediaRecognitionType.Dtmf` then we should not get back `choiceResult `object and in case of `CallMediaRecognitionType.Choices choiceResult` should not get  `collectTonesResult`

Reply:
right, it should be seperated.

Reply:
The event builder enforce the null value to  pick up,  which is not able to add null value in this case. Would keep all property  valued without null.